vmware-event-router/internal/processor/openfaas/openfaas.go

```diff
@@ -0,0 +1,322 @@
+package openfaas
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/avast/retry-go"
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	ofsdk "github.com/openfaas-incubator/connector-sdk/types"
+	"github.com/openfaas/faas-provider/auth"
+	"github.com/pkg/errors"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/vmware-samples/vcenter-event-broker-appliance/vmware-event-router/internal/color"
+	config "github.com/vmware-samples/vcenter-event-broker-appliance/vmware-event-router/internal/config/v1alpha1"
+	"github.com/vmware-samples/vcenter-event-broker-appliance/vmware-event-router/internal/metrics"
+	"github.com/vmware-samples/vcenter-event-broker-appliance/vmware-event-router/internal/processor"
+)
+
+const (
+	defaultTopicDelimiter  = ","
+	defaultRebuildInterval = time.Second * 10
+	defaultTimeout         = time.Second * 15
+)
+
+var (
+	// ErrStopped error is returned when the processor has been shutdown but there
+	// are still inflight processing requests
+	ErrStopped = errors.New("processor already stopped")
+
+	// assert we implement Processor interface
+	_ processor.Processor = (*Processor)(nil)
+)
+
+type logger interface {
+	Printf(format string, v ...interface{})
+	Println(v ...interface{})
+}
+
+// invokeFunc is a function which invokes the given OpenFaaS function (fn) with
+// the specified message body. It returns the function response message, status
+// code, http headers and error.
+type invokeFunc func(ctx context.Context, fn string, message []byte) ([]byte, int, http.Header, error)
```

Comment:
I think `fn` is a misleading name since it is a string, not a function type. Should the name be `fnName` or something like that?

Reply:
Changed.